core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushLimitThroughProject.java

```diff
@@ -42,6 +43,7 @@
     private static final Capture<ProjectNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
As long as the ordering doesn't depend on the outputs of the projection, pushing through a projection should work. It will need to rewrite the symbols in the ordering scheme, though